File: src/desktop-edge-agent/src/FccDesktopAgent.Core/Adapter/Doms/Model/DomsTransactionDto.cs
namespace FccDesktopAgent.Core.Adapter.Doms.Model;

/// <summary>
/// Raw DOMS transaction from the supervised buffer (FpSupTrans response).
/// Numeric values use integer types with explicit scale to avoid floating-point precision loss:
/// <list type="bullet">
///   <item><see cref="VolumeCl"/> — volume in centilitres</item>
///   <item><see cref="AmountX10"/> — x10 of minor currency units</item>
///   <item><see cref="UnitPriceX10"/> — x10 of minor currency units per litre</item>
/// </list>
/// </summary>
public sealed record DomsJplTransactionDto(
    string TransactionId,
    int FpId,
    int NozzleId,
    string ProductCode,
    long VolumeCl,
    long AmountX10,
    long UnitPriceX10,
    string Timestamp,
    string? AttendantId,
    int BufferIndex);

File: src/edge-agent/app/src/main/kotlin/com/fccmiddleware/edge/adapter/doms/model/DomsTransactionDto.kt
package com.fccmiddleware.edge.adapter.doms.model

import kotlinx.serialization.Serializable

/** Raw DOMS transaction data from the supervised buffer (FpSupTrans response). */
@Serializable
data class DomsTransactionDto(
    val transactionId: String,
    val fpId: Int,
    val nozzleId: Int,
    val productCode: String,
    /** Volume in centilitres (integer). */
    val volumeCl: Long,
    /** Amount in DOMS units (x10 of minor currency units). */
    val amountX10: Long,
    /** Unit price in DOMS units. */
    val unitPriceX10: Long,
    /** Timestamp in "yyyyMMddHHmmss" format, local time. */
    val timestamp: String,
    val attendantId: String? = null,
    val bufferIndex: Int,
)
